pinot-server/src/main/java/org/apache/pinot/server/starter/helix/SegmentOnlineOfflineStateModelFactory.java

```diff
@@ -215,6 +240,15 @@ public void onBecomeDroppedFromOffline(Message message, NotificationContext cont
       String segmentName = message.getPartitionName();
       try {
         _instanceDataManager.deleteSegment(tableNameWithType, segmentName);
+
+        // Check if the segment is recently offloaded from CONSUMING to OFFLINE
+        if (TableNameBuilder.isRealtimeTableResource(tableNameWithType)) {
+          Pair<String, String> tableSegmentPair = Pair.of(tableNameWithType, segmentName);
+          if (_recentlyOffloadedConsumingSegments.getIfPresent(tableSegmentPair) != null) {
+            _recentlyOffloadedConsumingSegments.invalidate(tableSegmentPair);
+            onConsumingToDropped(tableNameWithType, segmentName);
+          }
```

Comment:
This is not an actual problem, but it is a bit confusing. 

`_recentlyOffloadedConsumingSegments` is only populated with realtime tables, why do we need to check `if  (TableNameBuilder.isRealtimeTableResource(tableNameWithType))`?

Is the reason to do not create the pair?

Reply:
Yes, and to reduce a cache lookup. OFFLINE table doesn't really have CONSUMING state